Add Quick Start and Troubleshooting pages (#681)
reach-sh/reach-lang: docs/src/trouble/index.md

# {#ref-ts} Troubleshooting

If you receive one of the following errors: 

> 

Reviewer: You should make headings

```cmd
reach: /app/tmp/out.sh: openFile: permission denied (Permission denied)
```

OR

```
chmod: cannot access '/tmp/reach.../out.sh': No such file or directory
./reach: 1: /tmp/reach.../out.sh: not found
```

* This generally means that Reach was previously run or installed with `sudo` and directories were created that only `root` can access. 

* If this is the case, delete the local copy of the Reach program and reinstall per the instructions.

* If use of `sudo` during installation is not the cause then try [updating Docker](https://techdirectarchive.com/2021/10/17/how-to-manually-update-docker-desktop/) and Docker-compose.

"I'm running Windows 8 and cannot install Docker."

Unfortunately, Docker Desktop does not support Windows 8.

> Also, Reach is not able to run on Windows 8. 

Reviewer: s/Also/Thus

Possible solutions are:
* Upgrade to Windows 10 or Windows 11.
* Partition the drive (or use a second drive) and install of a copy of a supported Linux distro.
* Run a VM with a supported OS.

> It is likely that this option has the worst experience due to the splitting of system resources.

Reviewer: Indent

Executing `./reach version` on Linux outputs `./reach: Is a directory`

* This error usually occurs when `reach` is being called from a directory where it is not installed.

> Compare where Reach is installed with the current working directory. 

Reviewer: Indent

* Check the current working directory with the terminal command `pwd`.

## {#ref-ts-config} Reach configuration and REACH_CONNECTOR_MODE

> `Missing REACH_CONNECTOR_MODE`

Reviewer: I think that ``` will look better

* Reach expects to be told to connect to a specific consensus network.
Running ONE of the following commands should fix this error:

`export REACH_CONNECTOR_MODE=ALGO`

`export REACH_CONNECTOR_MODE=ETH`

`export REACH_CONNECTOR_MODE=CFX`

> 

Reviewer: We'd rather them use `reach config`

When executing `./reach config`, the compiler prints the following message:

```
Reach detected an existing configuration file at...

You appear to be using the `bash` shell, with environment configuration stored in /home/USER/.profile.
Run the following command to activate your new configuration:

$ . /home/USER/.profile
```

> * The compiler is asking you to execute the following command in your terminal. Note that `USER` designates your terminal's user name.

Reviewer: newline

`. /home/USER/.profile`

## {#ref-ts-vscode} Troubleshooting Reach VSCode Extension

"I don't see syntax highlighting in VSCode despite having the Reach extension installed and activated."

* Usually this is because VSCode needs to be restarted after installing the extension. 

* The extension is actively being updated.

> Sometimes after an update VSCode will need to be restarted in order for syntax highlighting to begin working again. 

Reviewer: Indent

Author: This shows in my rendering as inline with the previous sentence as a single bullet.

## {#ref-ts-permissions} Troubleshooting permissions

If running `./reach run` requires `sudo`, then it is likely that:

1. A component used by `reach` was installed using `sudo`, such as Docker. 

2. Docker might have been installed using Snap.

> Snap installation in Ubuntu creates permissions issues.

Reviewer: This whole thing is too vague and I think it should be in the first section

## {#ref-qs-mac-troubleshooting} MacOS troubleshooting

Receiving the following error on **M1 Macs**:

``` cmd
The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested`
```

> * Ensure Docker and Docker-compose are installed.

Reviewer: We know that it is just docker on this platform and we know from this error that docker _is_ installed... the error comes from Docker

* Check which mode the terminal is in by executing `$ arch` in the terminal.

> On M1 machines, the output should be `arm64`.

Reviewer: Indent

* Re-download the reach script in the desired directory:

> `$ curl https://docs.reach.sh/reach -o reach ; chmod +x reach`

Reviewer: ```cmd

Author: not sure what you are looking for here. Made some clarity edits.

Reviewer 2: I think Jay's asking for this to be a markdown code block rather than inline code (I concur if so).

Six tick marks instead of two.

It is likely that this option has the worst experience due to the splitting of system resources.

Executing `./reach version` on Linux outputs `./reach: Is a directory`

* This error usually occurs when `reach` is being called from a directory where it is not installed.
Compare where Reach is installed with the current working directory. 
* Check the current working directory with the terminal command `pwd`.

If running `./reach run` requires `sudo`, then it is likely that:

* A component used by `reach` was installed/created using `sudo`, such as Docker or the folder directory.
Make sure that all components are installed or created without using super-user permissions. 

* Docker might have been installed using Snap.
Snap installation in Ubuntu creates permissions issues.
Uninstall Docker and then re-install using the `apt` command in the terminal.

## {#ref-ts-config} Reach configuration and REACH_CONNECTOR_MODE

`Missing REACH_CONNECTOR_MODE`

* Reach expects to be told to connect to a specific consensus network.
Running ONE of the following commands should fix this error:

`export REACH_CONNECTOR_MODE=ALGO`

`export REACH_CONNECTOR_MODE=ETH`

`export REACH_CONNECTOR_MODE=CFX`

> When executing `./reach config`, the compiler prints the following message:

Reviewer 2: I appreciate what this section is trying to do given the surprising amount of confusion we've seen but I think you should actually just remove it:

- This isn't a sign of trouble
- The compiler isn't relevant here; I think you mean CLI
- It won't always be `~/.profile` - it depends on the user's `$SHELL` and how they've configured it, e.g. https://github.com/reach-sh/reach-lang/blob/8f9ceada61a0bf3a83d4de981014d4eb15f64c9e/hs/app/reach/Main.hs#L1708
- Right now I'm trying to add more padding around how `reach config` works anyway, so I think you should let me fail at that until I discover the least-failable way and we take this back out of confused users' hands

* If this is the case, delete the local copy of the Reach program and reinstall per the instructions.

* If use of `sudo` during installation is not the cause then try [updating Docker](https://techdirectarchive.com/2021/10/17/how-to-manually-update-docker-desktop/) and Docker-compose.

"I'm running Windows 8 and cannot install Docker."

Unfortunately, Docker Desktop does not support Windows 8.
Also, Reach is not able to run on Windows 8. 

Possible solutions are:
* Upgrade to Windows 10 or Windows 11.
* Partition the drive (or use a second drive) and install of a copy of a supported Linux distro.
* Run a VM with a supported OS.
It is likely that this option has the worst experience due to the splitting of system resources.

Executing `./reach version` on Linux outputs `./reach: Is a directory`

* This error usually occurs when `reach` is being called from a directory where it is not installed.
Compare where Reach is installed with the current working directory. 
* Check the current working directory with the terminal command `pwd`.

If running `./reach run` requires `sudo`, then it is likely that:

> * A component used by `reach` was installed/created using `sudo`, such as Docker or the folder directory.

Author: I disagree, because if you do `sudo mkdir`, you are creating a folder directory but you are not installing anything. I thought this would cause accessibility issues. Is that incorrect?

> * A component used by `reach` was installed/created using `sudo`, such as Docker or the folder directory.

Author: Oh, I gotcha. I thought you were saying the whole thing was redundant.

> * A component used by `reach` was installed/created using `sudo`, such as Docker or the folder directory.

Reviewer 2: Docker also _has_ to be installed by a user with elevated privileges, so I think this line is likely to just confuse people.

"I'm running Windows 8 and cannot install Docker."

Unfortunately, Docker Desktop does not support Windows 8.
Thus, Reach is not able to run on Windows 8. 

Possible solutions are:
* Upgrade to Windows 10 or Windows 11.
* Partition the drive (or use a second drive) and install a copy of a supported Linux distro.
* Run a VM with a supported OS.

  It is likely that running a VM has the worst experience due to the splitting of system resources.

Executing `./reach version` on Linux outputs `./reach: Is a directory`

* This error usually occurs when `reach` is being called from a directory where it is not installed.

  Compare where Reach is installed with the current working directory. 

* Check the current working directory with the terminal command `pwd`.

> If running `./reach run` seems to require `sudo`, then it is likely that:

Reviewer 2: ```suggestion
If running `./reach run` seems to require `sudo` it's possible one or more of the following is at issue:
```

> * A component used by `reach` was installed/created using `sudo`, such as the directory.

Reviewer 2: ```suggestion
* A directory used by `reach` was created with `sudo` or by another account.
```

  Make sure that the current user has read/write permissions in the directory. 

* Docker might have been installed using Snap.

  Snap installation in Ubuntu creates permissions issues.

> Uninstall Docker and then re-install using the `apt` command in the terminal.

Reviewer 2: ```suggestion
  Uninstall Docker and then re-install using the `apt` command in the terminal.
```
